File: WatcherF/src/shared/repositories/types/product.ts
export type ProductShort = {
  id: number;
  name: string;
  price: number;
  stockQuantity: number;
  pictureUrl: string;
  brand: IdWithName | null;
};

export type ProductFull = {
  id: number;
  name: string;
  description: string;
  country: IdWithName | null;
  price: number;
  stockQuantity: number;
  pictureUrl: string;
  brand: IdWithName | null;
  waterproof: number;
  glassType: IdWithType | null;
  caseType: IdWithType | null;
  styleType: IdWithStyle | null;
};

export type ProductListResponse = {
  content: Array<ProductShort>,
  page: number,
  size: number,
  totalElements: number,
  totalPages: number,
}

export type ProductListRequest = {
  style?: number;
  brand?: number;
  caseType?: number;
  glassType?: number;
  waterproofMin?: number;
  waterproofMax?: number;
  inStock?: boolean;
  priceMin?: number;
  priceMax?: number;
  page?: number;
  size?: number;
};

export type ProductUpdateRequest = {
  id: number;
  name?: string;
  description?: string;
  countryId?: number;
  price?: number;
  stockQuantity?: number;
  pictureUrl?: string;
  brandId?: number;
  waterproof?: number;
  glassTypeId?: number;
  caseTypeId?: number;
  styleTypeId?: number;
};

export type ProductCreateRequest = {
  name: string;
  description?: string;
  countryId?: number;
  price: number;
  stockQuantity: number;
  pictureUrl?: string;
  brandId?: number;
  waterproof?: number;
  glassTypeId?: number;
  caseTypeId?: number;
  styleTypeId?: number;
};



export type IdWithType = {
  id: number;
  type: string;
};
export type IdWithName = {
  id: number;
  type: string;
}
export type IdWithStyle = {
  id: number;
  style: string;
}

File: WatcherB/src/main/java/com/watcher/WatcherB/models/Data/Goods/Product.java
package com.watcher.WatcherB.models.Data.Goods;

import com.watcher.WatcherB.models.Data.Goods.Details.*;
import lombok.*;

import jakarta.persistence.*;
import java.math.BigDecimal;

@Entity
@Data
@NoArgsConstructor
@Table (name = "products")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private String name;

    private String description;

    @ManyToOne
    @JoinColumn(name = "country_id")
    private Country country;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(nullable = false)
    private Integer stockQuantity;

    private String pictureUrl;

    @ManyToOne
    @JoinColumn(name = "brand_id")
    private Brand brand;

    @Column(nullable = false)
    private Integer waterproof = 0;

    @ManyToOne
    @JoinColumn(name = "glass_type_id")
    private GlassType glassType;

    @ManyToOne
    @JoinColumn(name = "case_type_id")
    private CaseType caseType;

    @ManyToOne
    @JoinColumn(name = "style_id")
    private StyleType styleType;
}
